civis/io/_tables.py

```diff
@@ -1316,168 +1299,148 @@ def _run_cleaning(file_ids, client, need_table_columns, headers, delimiter,
     return cleaning_futures
 
 
-def _check_all_detected_info(detected_info, headers, delimiter,
-                             compression, output_file_id):
-    """Check a single round of cleaning results as compared to provided values.
-
-    Parameters
-    ----------
-    detected_info: Dict[str, Any]
-      The detected info of the file as returned by the Civis API.
-    headers: bool
-      The provided value for whether or not the file contains errors.
-    delimiter: str
-      The provided value for the file delimiter.
-    compression: str
-      The provided value for the file compression.
-    output_file_id: int
-      The cleaned file's Civis ID. Used for debugging.
-
-    Raises
-    ------
-    CivisImportError
-      If the values detected on the file do not match their expected
-      attributes.
-    """
-    if headers != detected_info['includeHeader']:
-        raise CivisImportError('Mismatch between detected headers - '
-                               'please ensure all imported files either '
-                               'have a header or do not.')
-    if delimiter != detected_info['columnDelimiter']:
-        raise CivisImportError('Provided delimiter "{}" does not match '
-                               'detected delimiter for {}: "{}"'.format(
-                                    delimiter,
-                                    output_file_id,
-                                    detected_info["columnDelimiter"])
-                               )
-    if compression != detected_info['compression']:
-        raise CivisImportError('Mismatch between detected and provided '
-                               'compressions - provided compression was {}'
-                               ' but detected compression {}. Please '
-                               'ensure all imported files have the same '
-                               'compression.'.format(
-                                   compression,
-                                   detected_info['compression'])
-                               )
-
-
 def _process_cleaning_results(cleaning_futures, client, headers,
                               need_table_columns, delimiter):
```

Comment:
I ended up having to rewriting `_process_cleaning_results` and its dependent helper functions. Before this PR, the logic of `_processing_cleaning_results` was to use the first input file as the reference and then check the second and remaining files against this first file for type consistency. Simply rewriting the helper function `_check_column_types` that checked two files' type info wouldn't be sufficient, because it'd still crash for a situation like three input files with [INT, FLOAT, VARCHAR] for a given column, where checking (INT, FLOAT) would raise an error before the code would know about a possible upcoming VARCHAR. For what we want now, we need to collect all the detected SQL types across all the files before we can apply the checking + relaxing logic.